gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/documentation/domain_service/ApiDocumentationDomainService.java

```diff
@@ -15,25 +15,54 @@
  */
 package io.gravitee.apim.core.documentation.domain_service;
 
+import io.gravitee.apim.core.api.crud_service.ApiCrudService;
+import io.gravitee.apim.core.documentation.crud_service.PageCrudService;
 import io.gravitee.apim.core.documentation.model.Page;
 import io.gravitee.apim.core.documentation.query_service.PageQueryService;
+import io.gravitee.apim.core.exception.DomainException;
 import java.util.List;
 import java.util.Objects;
 
 public class ApiDocumentationDomainService {
 
+    private static final String ROOT = "ROOT";
     private final PageQueryService pageQueryService;
+    private final PageCrudService pageCrudService;
+    private final ApiCrudService apiCrudService;
 
-    public ApiDocumentationDomainService(PageQueryService pageQueryService) {
+    public ApiDocumentationDomainService(
+        PageQueryService pageQueryService,
+        PageCrudService pageCrudService,
+        ApiCrudService apiCrudService
+    ) {
         this.pageQueryService = pageQueryService;
+        this.pageCrudService = pageCrudService;
+        this.apiCrudService = apiCrudService;
     }
 
     public List<Page> getApiPages(String apiId, String parentId) {
         if (Objects.nonNull(parentId) && !parentId.isEmpty()) {
-            var parentIdParam = "ROOT".equals(parentId) ? null : parentId;
+            var parentIdParam = ROOT.equals(parentId) ? null : parentId;
             return pageQueryService.searchByApiIdAndParentId(apiId, parentIdParam);
         } else {
             return pageQueryService.searchByApiId(apiId);
         }
     }
+
+    public Page getPageAndVerifyApi(String apiId, String pageId, boolean ignoreRoot) {
```

Comment:
Suggestion: what about splitting this method in 2 `checkApiExists` and `checkPageExists` ?

I am a bit disturbed by the `ignoreRoot` parameter and I feel it's more that we are doing 2 separate things in the same method. WDYT ?

Reply:
I've split this up so that the check for if the `Api` and `Page` exist is in each use-case directly. I have only included the check for if the page is associated to a given `Api` in the `ApiDocumentationDomainService`